cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/client/AzureClient.java

```diff
@@ -284,16 +287,25 @@ public void deleteBlobInStorageContainer(String resourceGroup, String storageNam
         }
     }
 
-    public Disk createManagedDisk(String diskName, int diskSize, AzureDiskType diskType, String region, String resourceGroupName, Map<String, String> tags) {
+    public Disk createManagedDisk(String diskName, int diskSize, AzureDiskType diskType, String region, String resourceGroupName, Map<String, String> tags,
+            String diskEncryptionSetId) {
         LOGGER.debug("create managed disk with name={}", diskName);
-        return azure.disks().define(diskName)
+        Disk.DefinitionStages.WithCreate withCreate = azure.disks().define(diskName)
                 .withRegion(Region.findByLabelOrName(region))
                 .withExistingResourceGroup(resourceGroupName)
                 .withData()
                 .withSizeInGB(diskSize)
                 .withTags(tags)
-                .withSku(convertAzureDiskTypeToDiskSkuTypes(diskType))
-                .create();
+                .withSku(convertAzureDiskTypeToDiskSkuTypes(diskType));
+        if (diskEncryptionSetId != null) {
```

Comment:
extract this please

Reply:
Done, see `setupDiskEncryptionWithDesIfNeeded()`.